### ISSUE TITLE
additionalProperties in items

### ISSUE THREAD

**Issue by holgpar** [harrel56/json-schema#109]
Hi,

I have encountered the following behavior. For a schema:

```json
{
    "type": "object",
    "properties" : {
        "list": {
            "type": "array",
            "items": {
                "type": "object",
                "properties": {
                    "x1": {
                        "type": "string"
                    }
                },
                "additionalProperties": { "type": "number" }
            }
        }
    }
}
```

I try to validate a document:

```json
 {
         "list": [ { }, { "x1" : "foo"}  ]
 }
```

This fails with:

> [ {
  "evaluationPath" : "/properties/list/items/additionalProperties/type",
  "schemaLocation" : "https://harrel.dev/e6182f0b#/properties/list/items/additionalProperties",
  "instanceLocation" : "/list/1/x1",
  "keyword" : "type",
  "error" : "Value is [string] but should be [number]"
} ]

If I remove the first element of "list" it passes.

**Comment by harrel56** [comment]
Yeah, this indeed looks strange. Yet I might have an idea what's causing this - gonna try to fix it later. Thanks for reporting it!

**Comment by holgpar** [comment]
Thanks for looking into it.

I found another issue which is probably related. Given the schema

```json
{
    "type": "object",
    "properties": {
        "x": {
            "type": "object",
            "properties": {
                "a": {
                    "type": "string"
                }
            },
            "additionalProperties": false
        },
        "y": {
            "type": "object",
            "properties": {
                "a": {
                    "type": "string"
                }
            },
            "additionalProperties": false
        }
    }
}
```

the input

```json
  {
             "x": {}, "y": { "a" : "foo" }
   }
```

comes back with:

```json
[ {
  "evaluationPath" : "/properties/y/additionalProperties",
  "schemaLocation" : "https://harrel.dev/2c848018#/properties/y/additionalProperties",
  "instanceLocation" : "/y/a",
  "keyword" : null,
  "error" : "False schema always fails"
} ]
```

And again, if I remove the "x", it it passes.

**Comment by holgpar** [comment]
I thought the issue would be that there is a property with the same name ("a") in two subobjects/items, but that does not matter:

```json
{
    "type": "object",
    "properties": {
        "y": {
            "type": "object",
            "properties": {
                "a": {
                    "type": "string"
                }
            },
            "additionalProperties": false
        },
        "z": {
            "type": "object",
            "properties": {
                "b": {
                    "type": "string"
                }
            },
            "additionalProperties": false
        }
    }
}
```

fails:

```json
  {
                             "y": { "a": "foo"}, "z": { "b" : "foo" }
  }
```

**Comment by harrel56** [comment]
@holgpar Just released version `1.5.0` with the fixes (and also format validation support). Thanks again and feel free to reopen or file new issue if something is still not working for you.
